gcloud-java-storage/README.md

```diff
@@ -56,29 +56,160 @@ Cloud Storage for your project.
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageOptions;
+
+Storage storage = StorageOptions.defaultInstance().service();
+```
+
+For other authentication options, see the [Authentication](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) page.
+
+#### Storing data
+Stored objects are called "blobs" in `gcloud-java` and are organized into containers called "buckets".  In this code snippet, we will create a new bucket and upload a blob to that bucket.
+
+Add the following imports at the top of your file:
 
 ```java
 import static java.nio.charset.StandardCharsets.UTF_8;
 
+import com.google.gcloud.storage.BlobId;
+import com.google.gcloud.storage.BlobInfo;
+import com.google.gcloud.storage.BucketInfo;
+```
+
+Then add the following code to create a bucket and upload a simple blob.
+
+*Important: Bucket names have to be globally unique. If you choose a bucket name that already exists, you'll get a helpful error message telling you to choose another name. In the code below, replace "my_unique_bucket" with a unique bucket name.*
+
+```java
+// Create a bucket
+String bucketName = "my_unique_bucket"; // Remember to change this to something unique
+BucketInfo bucketInfo = storage.create(BucketInfo.builder(bucketName).build());
+
+// Upload a blob to the newly created bucket
+BlobId blobId = BlobId.of(bucketName, "my_blob_name");
+BlobInfo blobInfo = storage.create(BlobInfo.builder(blobId).build(),
+                                   "a simple blob".getBytes(UTF_8));
+```
+
+At this point, you will be able to see your newly created bucket and blob on the Google Developers Console.
+
+#### Retrieving data
+Now that we have content uploaded to the server, we can see how to read data from the server. Add the following import:
+
+```java
 import com.google.gcloud.storage.Blob;
-import com.google.gcloud.storage.Storage;
-import com.google.gcloud.storage.StorageOptions;
+```
 
-import java.nio.ByteBuffer;
-import java.nio.channels.WritableByteChannel;
+Then add the following lines to your program to get back the blob we uploaded.
 
-Storage storage = StorageOptions.defaultInstance().service();
-Blob blob = new Blob(storage, "bucket", "blob_name");
-if (!blob.exists()) {
-  storage2.create(blob.info(), "Hello, Cloud Storage!".getBytes(UTF_8));
-} else {
-  System.out.println("Updating content for " + blob.info().name());
-  byte[] prevContent = blob.content();
-  System.out.println(new String(prevContent, UTF_8));
-  WritableByteChannel channel = blob.writer();
-  channel.write(ByteBuffer.wrap("Updated content".getBytes(UTF_8)));
-  channel.close();
+```java
+Blob blob = Blob.load(storage, blobInfo.blobId());
+String blobContent = new String(blob.content(), UTF_8);
+```
+
+If others have permission to edit the blob, then you may want to call `reload` to get a more up to date copy of the blob later in your program. The following snippet shows how to get a new Blob object containing updated information.
```

Comment:
I am not sure we need this section,  however if you feel we should keep it I think we should
just say something like `to get a fresh state of your blob's metadata you can use...`